### ISSUE TITLE
Make -j4 returns an error.

### ISSUE THREAD

**Issue by ebasatemesgen** [Omegastick/pytorch-cpp-rl#19]
hello, when every I try to 'make -j4' it returns this error, I don't know how to get around it, can you please help me with it. Thank you.

[  0%] Building C object example/lib/libzmq/CMakeFiles/objects.dir/src/tweetnacl.c.o
[  1%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/precompiled.cpp.o
[  1%] Building CXX object CMakeFiles/cpprl_tests.dir/src/storage.cpp.o
[  1%] Building CXX object CMakeFiles/cpprl.dir/src/storage.cpp.o
[  2%] Building CXX object CMakeFiles/cpprl.dir/src/observation_normalizer.cpp.o
[  3%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/address.cpp.o
[  3%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/client.cpp.o
[  4%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/clock.cpp.o
[  4%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/ctx.cpp.o
[  5%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/curve_mechanism_base.cpp.o
[  6%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/curve_client.cpp.o
[  6%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/curve_server.cpp.o
[  7%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/dealer.cpp.o
[  7%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/devpoll.cpp.o
[  8%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/dgram.cpp.o
[  9%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/dist.cpp.o
[  9%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/endpoint.cpp.o
[ 10%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/epoll.cpp.o
[ 11%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/err.cpp.o
[ 11%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/fq.cpp.o
[ 12%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/io_object.cpp.o
[ 12%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/io_thread.cpp.o
[ 13%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/ip.cpp.o
[ 14%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/ipc_address.cpp.o
[ 14%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/ipc_connecter.cpp.o
[ 15%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/ipc_listener.cpp.o
[ 16%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/kqueue.cpp.o
[ 16%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/lb.cpp.o
[ 17%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/mailbox.cpp.o
[ 17%] Building CXX object CMakeFiles/cpprl.dir/src/running_mean_std.cpp.o
[ 18%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/mailbox_safe.cpp.o
[ 18%] Building CXX object CMakeFiles/cpprl.dir/src/algorithms/a2c.cpp.o
[ 19%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/mechanism.cpp.o
[ 20%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/mechanism_base.cpp.o
[ 20%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/metadata.cpp.o
[ 21%] Building CXX object CMakeFiles/cpprl_tests.dir/src/observation_normalizer.cpp.o
[ 22%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/msg.cpp.o
[ 23%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/mtrie.cpp.o
[ 23%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/norm_engine.cpp.o
[ 24%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/object.cpp.o
[ 24%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/options.cpp.o
[ 25%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/own.cpp.o
[ 26%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/null_mechanism.cpp.o
[ 26%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/pair.cpp.o
[ 27%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/pgm_receiver.cpp.o
[ 28%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/pgm_sender.cpp.o
[ 28%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/pgm_socket.cpp.o
[ 29%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/pipe.cpp.o
[ 29%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/plain_client.cpp.o
[ 30%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/plain_server.cpp.o
[ 31%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/poll.cpp.o
[ 31%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/poller_base.cpp.o
[ 32%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/polling_util.cpp.o
[ 32%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/pollset.cpp.o
[ 33%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/proxy.cpp.o
[ 34%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/pub.cpp.o
[ 34%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/pull.cpp.o
[ 35%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/push.cpp.o
[ 36%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/random.cpp.o
[ 36%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/raw_encoder.cpp.o
[ 37%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/raw_decoder.cpp.o
[ 37%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/reaper.cpp.o
[ 38%] Building CXX object CMakeFiles/cpprl.dir/src/algorithms/ppo.cpp.o
[ 39%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/rep.cpp.o
[ 39%] Building CXX object CMakeFiles/cpprl.dir/src/distributions/bernoulli.cpp.o
[ 40%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/req.cpp.o
[ 40%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/router.cpp.o
[ 41%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/select.cpp.o
[ 42%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/server.cpp.o
[ 42%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/session_base.cpp.o
[ 43%] Building CXX object CMakeFiles/cpprl_tests.dir/src/running_mean_std.cpp.o
[ 44%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/signaler.cpp.o
[ 44%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/socket_base.cpp.o
/home/kbots/eba/pytorch-cpp-rl/src/distributions/bernoulli.cpp: In member function ‘virtual at::Tensor cpprl::Bernoulli::entropy()’:
/home/kbots/eba/pytorch-cpp-rl/src/distributions/bernoulli.cpp:46:101: error: ‘Reduction’ has not been declared
 ntropy_with_logits(logits, probs, torch::Tensor(), torch::Tensor(), Reduction::None);
                                                                     ^~~~~~~~~

/home/kbots/eba/pytorch-cpp-rl/src/distributions/bernoulli.cpp: In member function ‘virtual at::Tensor cpprl::Bernoulli::log_prob(at::Tensor)’:
/home/kbots/eba/pytorch-cpp-rl/src/distributions/bernoulli.cpp:52:135: error: ‘Reduction’ has not been declared
 nsors[0], broadcasted_tensors[1], torch::Tensor(), torch::Tensor(), Reduction::None);
                                                                     ^~~~~~~~~

[ 45%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/socks.cpp.o
[ 46%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/socks_connecter.cpp.o
[ 46%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/stream.cpp.o
[ 47%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/stream_engine.cpp.o
[ 48%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/sub.cpp.o
make[2]: *** [CMakeFiles/cpprl.dir/build.make:128: CMakeFiles/cpprl.dir/src/distributions/bernoulli.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 48%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/tcp.cpp.o
[ 48%] Building CXX object CMakeFiles/cpprl_tests.dir/src/algorithms/a2c.cpp.o
[ 50%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/tcp_address.cpp.o
[ 50%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/tcp_connecter.cpp.o
[ 51%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/tcp_listener.cpp.o
[ 52%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/thread.cpp.o
[ 52%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/trie.cpp.o
[ 53%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/radix_tree.cpp.o
[ 53%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/v1_decoder.cpp.o
[ 54%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/v1_encoder.cpp.o
make[1]: *** [CMakeFiles/Makefile2:73: CMakeFiles/cpprl.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 55%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/v2_decoder.cpp.o
[ 56%] Building CXX object CMakeFiles/cpprl_tests.dir/src/algorithms/ppo.cpp.o
[ 56%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/v2_encoder.cpp.o
[ 57%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/xpub.cpp.o
[ 58%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/xsub.cpp.o
[ 58%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/zmq.cpp.o
[ 59%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/zmq_utils.cpp.o
[ 59%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/decoder_allocators.cpp.o
[ 60%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/socket_poller.cpp.o
[ 61%] Building CXX object CMakeFiles/cpprl_tests.dir/src/distributions/bernoulli.cpp.o
[ 62%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/timers.cpp.o
[ 62%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/radio.cpp.o
[ 63%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/dish.cpp.o
[ 64%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/udp_engine.cpp.o
[ 64%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/udp_address.cpp.o
[ 65%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/scatter.cpp.o
[ 65%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/gather.cpp.o
[ 66%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/ip_resolver.cpp.o
/home/kbots/eba/pytorch-cpp-rl/src/distributions/bernoulli.cpp: In member function ‘virtual at::Tensor cpprl::Bernoulli::entropy()’:
/home/kbots/eba/pytorch-cpp-rl/src/distributions/bernoulli.cpp:46:101: error: ‘Reduction’ has not been declared
     return torch::binary_cross_entropy_with_logits(logits, probs, torch::Tensor(), torch::Tensor(), Reduction::None);
                                                                                                     ^~~~~~~~~
/home/kbots/eba/pytorch-cpp-rl/src/distributions/bernoulli.cpp: In member function ‘virtual at::Tensor cpprl::Bernoulli::log_prob(at::Tensor)’:
/home/kbots/eba/pytorch-cpp-rl/src/distributions/bernoulli.cpp:52:135: error: ‘Reduction’ has not been declared
     return -torch::binary_cross_entropy_with_logits(broadcasted_tensors[0], broadcasted_tensors[1], torch::Tensor(), torch::Tensor(), Reduction::None);
                                                                                                                                       ^~~~~~~~~
[ 67%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/zap_client.cpp.o
[ 67%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/stream_connecter_base.cpp.o
[ 68%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/stream_listener_base.cpp.o
[ 69%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/tipc_address.cpp.o
[ 69%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/tipc_connecter.cpp.o
[ 70%] Building CXX object example/lib/libzmq/CMakeFiles/objects.dir/src/tipc_listener.cpp.o
[ 70%] Building CXX object CMakeFiles/cpprl_tests.dir/src/distributions/categorical.cpp.o
[ 70%] Built target objects
[ 71%] Building CXX object CMakeFiles/cpprl_tests.dir/src/distributions/distribution.cpp.o
make[2]: *** [CMakeFiles/cpprl_tests.dir/build.make:128: CMakeFiles/cpprl_tests.dir/src/distributions/bernoulli.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:142: CMakeFiles/cpprl_tests.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
